README.md

```diff
@@ -104,70 +176,132 @@ Username:
+
+Ensure your certificates cover the following subdomains at a bare minimum:
+
+- `opal.<domain>`
+- `keycloak.<domain>`
+- `minio.<domain>`
+
+#### For Local Deployments (must be internet-connected):
+
+We have provided a tool for generating local selfsigned certificates that cover all opal subdomains. To generate these certificates:
+
+1. `cd` into `k8s-utils/cert-gen`
+2. Run `bash gen_selfsigned_certs.bash`. This will create the files `tls.crt` and `tls.key`
+
+#### Install Certificates
+
+Place your certificates in `opal-helm/opal-setup`, naming your certificate and private key `tls.crt` and `tls.key`, respectively.
+
+### Image Registry Authentication
+
+In all cases but one (busybox...), OPAL uses images stored in Platform1's IronBank. Assuming you're not mirroring these images already for an airgapped deployment, you will need to provide your docker credentials to allow kubernetes to download them.
 
-First close down the port-forwarding service by closing the terminal or ctl+c This will cause the browser URL to stop working.
+#### Install dockerconfig.json
 
-### TLS certs
-Supply TLS certs in the same directory as opal-helm/opal-setup
+To allow kubernetes to authenticate with a private image registry, you'll first need to log in to that registry through the docker cli.
+In the case of IronBank, this will be `docker login registry1.dso.mil`. If using a different private registry, use that URL instead.
 
-### Add docker configuration JSON
-In order for the helm install to work properly in a local instance, it is necessary to add the docker config json file in the opal-helm/opal-setup directory.  This is needed for the helm chart to pull the appropriate image from a private repository.
+Upon successful login, assuming you are using a Linux system, the login command should have created the file `~/.docker/config.json`. Copy this to `opal-helm/opal-setup`, and rename it to `dockerconfig.json`
 
 ### Git Sync
+
 If enabled in the values.yaml, jupyterhub will add sidecars containers to the hub pod that will periodically pull in updates from the `opal` and `weave` repositories. 
-**NOTE**: A github account and access token with repo scope is currently required to use this functionality.
+
+> **NOTE**: Some repositories may require a github account and access token with repo scope.
 
 To use git-sync, place your token in opal-setup, and rename it to `git-creds`. In opal-setup/values, fill in your username in jupyterhub:appValues:gitSync
 
-### Helm install OPAL
+### Install the OPAL helm chart
+
+Prior to installing OPAL, you will likely need to change your working branch in git. At the time of writing, the current production branch is `aks-comp-rebase-test`. If you are testing a specific branch or have configurations stored in another branch, use that instead. Additionally, ensure that the value of `targetRevision` in `opal-helm/opal-setup/values.yaml` is set to the correct branch, or you may run into unforeseen issues.
```

Comment:
I'm cautious that: the production version:
-  is a branch not a tag
- is not the main, master or vX.Y branch
- is named  `...-test`.

If this is correct, I would think it needs explaining here vs. just saying to use that branch

Reply:
This branch got so far ahead of main and had so many differing implementations at one point that it was easier to just use this as the new main. We now have a live environment running off of it and I don't want anything to break, so it just stuck like that.

This chart is also still in its infancy. There aren't official releases, and the live server exists as a proof of concept that largely goes unused.

Reply:
I do plan to eventually get this all into the actual main branch, but it's unfortunately quite low in priorities over our other tasking

Reply:
I guess ignore the previous comment. I've archived the old main and copied/renamed rebase to make things less confusing